src/Application.php

```diff
@@ -217,195 +228,50 @@ public function getAdapterFactory()
-                            'tracker.';
-
-                        throw new \RuntimeException(sprintf($message, get_class($this->adapter)));
-                    }
-                }
-            }
-        }
-
-        /** @var \Symfony\Component\Console\Helper\Helper[] $helperSet */
+        /** @var \Symfony\Component\Console\Helper\HelperSet $helperSet */
         $helperSet = $command->getHelperSet();
+
         foreach ($helperSet as $helper) {
             if ($helper instanceof OutputAwareInterface) {
                 $helper->setOutput($output);
             }
-        }
-
-        parent::doRunCommand($command, $input, $output);
-    }
-
-    /**
-     * @return Adapter
-     */
-    private function determineAdapter()
-    {
-        $builder = new ProcessBuilder(['git', 'config', '--get', 'remote.origin.url']);
-        $builder
-            ->setWorkingDirectory(getcwd())
-            ->setTimeout(3600)
-        ;
-
-        $process = $builder->getProcess();
-        $process->run();
-
-        if (!$process->isSuccessful()) {
-            throw new \RuntimeException(
-                'The adapter type could not be determined (no Git origin configured for this repository). '.
-                'Please run the "core:init" command.'
-            );
-        }
-
-        $remoteUrl = strtolower($process->getOutput());
-        $ignoredAdapters = [];
-
-        foreach (array_keys($this->getAdapterFactory()->all()) as $adapterName => $adapterInfo) {
-            $config = $this->config->get(sprintf('[adapters][%s]', $adapterName));
-
-            // Adapter is not configured ignore
-            if (null === $config) {
-                $ignoredAdapters[] = $adapterName;
-
-                continue;
-            }
 
-            if (!$adapterInfo[AdapterFactory::SUPPORT_REPOSITORY_MANAGER]) {
-                continue;
+            if ($helper instanceof InputAwareInterface) {
+                $helper->setInput($input);
             }
-
-            $adapter = $this->adapterFactory->createRepositoryManager(
-                $adapterName,
-                $config,
-                $this->config
-            );
-
-            if ($adapter->supportsRepository($remoteUrl)) {
-                return $adapter;
-            };
         }
 
-        $exceptionMessage = 'The adapter type could not be determined.';
+        $event = new ConsoleCommandEvent($command, $input, $output);
+        $this->dispatcher->dispatch(ConsoleEvents::COMMAND, $event);
 
-        if ([] !== $ignoredAdapters) {
-            $exceptionMessage .= sprintf(
-                'Note, the following adapters (may support this repository) but are currently not configured: "%s".',
-                implode('", "', $ignoredAdapters)
-            );
+        try {
+            $exitCode = $command->run($input, $output);
+        } catch (\Exception $e) {
+            $event = new ConsoleTerminateEvent($command, $input, $output, $e->getCode());
```

Comment:
This can be added to a `finally` after the `try...catch`, then you don't need to duplicate the code to dispatch this event, and the terminate event fires after the exception

```php
try {
    $exitCode = $command->run($input, $output);
} catch (\Exception $e) {
    // dispatch exception event
} finally {
 // Dispatch console terminate event
}